### ISSUE TITLE
Vuze/BiglyBT keep crashing

### ISSUE THREAD

**Issue by magus186** [BiglySoftware/BiglyBT#1270]
Hey guys, have a problem I haven't been able to solve. I recently built a new PC with an i5-9400f and running Windows 7 on it, and have migrated all of my appdata from previous install of W7 to the new installation. However, it keeps crashing after 20 seconds or even 15 minutes, the times are random. IE is running V11 from windows update already. There is no error which appears on the screen, it just disappears and closes. I have followed the whole list from "https://wiki.vuze.com/w/Vuze_disappears" but nothing has helped. I have no AV as of yet on this fresh install of W7 (except malwarebytes) and everything else is running properly. Defender was deactivated. I even downgraded my JRE on windows to the same version I had on the previous install but no change. I've tried everything I could think of, including running fresh installations without migrating the appdata, and even that crashes. Also, this issue is happening to me with BOTH Bigly 2.1 AND Vuze 5.7.6. They are both having the exact same issues. Hoping you can give me some guidance to correct this, or if there are any windows updates that may be interfering etc. I've searched the internet long and hard for a solution without luck and as an avid user of Azureus from 2013 I'd like to get this running again! Thank you for your time. 

* Windows 7 x64 Pro
* BiglyBT 2.1.0.0

**Comment by parg** [comment]
Hi, sorry you're encountering this problem :( Are there any crash files generated by the Java runtime? They have names of the form "hs_err_pid<nnnn>.log" although off the top of my head I can't remember where they might be located.

You could also try generating a debug.zip file (Help menu, last option if it stays up long enough) and email that over to paul@biglybt.com

cheers!

**Comment by magus186** [comment]
> paul@biglybt.com

Hi Parg, thank you very much for the reply! I've just sent you an email with both debug files. I saw the java dump files you mention "hs_err_pidxxxx.log" as part of the debug files, so hopefully that will be enough? Let me know. I think anyway both softwares are using the internal java builds and not the one I had installed.

**Comment by NaPurHab** [comment]
Given the CPU you are running, you really should be on Windows 10. Support for Windows 7 ends in January 2020, in 4 months.

As to Java, the you can get the latest version [Here](https://www.java.com/en/download/win10.jsp).
But to debug if it is a Java issue, try these steps.

1. Uninstall Java using the Windows Settings Apps & Features.
2. Open a command prompt and type 'Java'. If there is anything still running you should see a result.
3. Install Java from [Here](https://www.java.com/en/download/win10.jsp).
4 Open Command prompt and type 'Java -version'    You should see the following:
`java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.221-b11, mixed mode)`

**Comment by parg** [comment]
All those crashes in the JVM at random places make me think of other instances like this caused by in-process memory corruption as the result of some dodgy DLL injection. Various applications (Alcohol 100 comes to mind but there are a lot of others listed on that page you referenced - https://wiki.vuze.com/w/Vuze_disappears ) stick their dirty fingers into processes to do whatever it is they do and if they have a bug they can cause random crashes like this. 

That being said, I looked at the DLLs listed as loaded in the dumps you sent and nothing obvious stands out :( Have you installed anything that you think might require such process meddling functionality to work?

**Comment by parg** [comment]
Hmm, that being said all the cashes are in thread "Cn CompilerThreadm" which goes against my random memory corruption theory.

There is a more recent version of Java 8 - 8u221 - https://www.java.com/en/download/faq/release_changes.xml

Might be worth trying that out

**Comment by magus186** [comment]
Thanks guys for the tips, definitely something going on with Java as mentioned

First thing, I did as suggested by NaPurHab and uninstalled Java, Vuze, and Bigly. First reinstalled Java, then Vuze, then BiglyBT. Java is showing updated to latest 

"C:\Users\Owner>java -version
java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.221-b11, mixed mode)"

And showing in "about" in Vuze and Bigly correctly now

Vuze: Java 1.8.0_221 (64 bit)
  Oracle Corporation
c:\program files\java\jre1.8.0_221

SWT v4716, win32
Windows 7 v6.1, amd64 (64 bit)
V5.7.6.0/4 az2

Bigly: 
Java 1.8.0_221 (64 bit)
  Oracle Corporation
c:\program files\java\jre1.8.0_221

SWT v4922, win32, zoom=100, dpi=96
Windows 7 v6.1, amd64 (64 bit)
B2.1.0.0/4 az2

But, the crashes persist :-(

A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x0000000062d0f4f8, pid=7992, tid=0x0000000000001f04
#
# JRE version: Java(TM) SE Runtime Environment (8.0_221-b11) (build 1.8.0_221-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.221-b11 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# V  [jvm.dll+0x4bf4f8]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp

Strange thing is, I use JDownloader also which is also running on Java, and that is completely stable..Vuze/Bigly crashing has nothing to do whether JDownloader is running or not though, have tried the variables. 

The Windows 7 I'm using now was freshly installed and updated. I don't run Alcohol either. Are there any other suggestions of something to try...? :-/

**Comment by parg** [comment]
I'm running Windows 7 too so don't think that is anything to do with things. Have you tried a more recent Java version? Version 13 has just been released:

https://www.oracle.com/technetwork/java/javase/downloads/index-jsp-138363.html#javasejdk

**Comment by magus186** [comment]
Thank you for the suggestion...I just tried that SE 13, but it seems that even though I downloaded and installed the SE, when installing Vuze/Bigly they both defaulted back to the built-in versions of Java RE.. I looked and couldn't find an RE version 13, is there something I'm missing or doing wrong..?

**Comment by parg** [comment]
hmm, try deleting the 'jre' folder in C:\Program Files\BiglyBT - maybe it'll then pick up the newer Java

**Comment by magus186** [comment]
I *think* you may have found the solution here! So far it's running for 25 minutes without crashing now! I will update either way if it crashes or not. 

Now I have another question in the meantime, is there any file i can edit to allow vuze to use the Java13 too? When I did the same process there it came back saying "No JVM could be found on your system. Please define EXE4J_JAVA_HOME to point to an installed 64-bit JDK"
Unfortunately one of the trackers I use doesn't support Bigly (yet), if that would be possible to redirect too you'd really be a lifesaver for me!

**Comment by parg** [comment]
Have a look at

https://wiki.vuze.com/w/Commandline_options#Manually_starting_directly_with_Java

With luck you'll be able to figure it out!

**Comment by magus186** [comment]
Thank you parg, it worked :-) I had to change the second line of code just a little to "C:\Program Files\Java\jdk-13\bin\java.exe" just for reference if anybody else faces the same issue as I have. 

I do face a minor annoyance though (may not be possible to fix in Vuze) but I can't open magnets or torrent files with Vuze Launcher. 

Anyway one last question, what is your paypal so I can donate to you for your help? :-)

**Comment by parg** [comment]
Can you live with copy-pasting the .torrent file/magnet links into the Vuze 'search bar' at the top and hitting 'enter'? Alternatively you could set a system property for EXE4J_JAVA_HOME - go to Control Panel->System, click on 'Advanced system settings', select the bottom button labeled 'Environment Variables' and then under 'User variables for <name>' select the New button and enter

EXE4J_JAVA_HOME

as the name and

C:\Program Files\Java\jdk-13\bin\java.exe

as the value. The value might have to have " around it, not sure.

Log out and in again and see if Vuze.exe now runs.

Thanks for persevering with this problem! I'm a little worried that we never figured out a cause for your problem other than a Java8->13 upgrade seems to have fixed it. We were hoping to stick with Java8 as our recommended JRE for a bit longer.

Thanks for the offer of a donation - if you can spare a few pennies then please use the 'donate' link in the Help menu in BiglyBT - there's a PayPal option there.

Cheers!

**Comment by magus186** [comment]
Thank you again parg! I just created the Enviornment Variables, had to adjust the value a little here to C:\Program Files\Java\jdk-13 only, without any quotation marks, and I added on both system and user (not sure if that's necessary or not but I did anyway). Restarted and now everything works great and perfect :-) 

I'm not sure if this is such a widespread problem, or as you say what was causing it or not, but hopefully for a majority of users Java8 is still sufficient. However at least we have a working idea now how to correct in case Java13 is needed for some reason (driver/windows update conflict?). 

Likewise thank you for persevering with my rare issue as well and helping me to solve it! I will definitely make a generous donation shortly for your help. 

Thank you!
